File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 1:
            return 0

        left = 0
        right = 1
        profit = 0

        while right < len(prices):
            if prices[right] - prices[left] < 0:
                left = right
                right += 1
            
            else:
                if prices[right] - prices[left] > profit:
                    profit = prices[right] - prices[left]
                right += 1

        return profit

File: Data Structures & Algorithms/palindromic-substrings/submission-1.java
class Solution {
    private int res = 0;
    public int countSubstrings(String s) {
        int n = s.length();

        for (int i = 0; i < n - 1; i++) {
            check(s, i, i);
            check(s, i, i + 1);
        }

        return res + 1;
    }

    private void check(String s, int left, int right) {
        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)) {
            res += 1;
            left -= 1;
            right += 1;
        }

        return;
    }
}
